tests/test_modeling_tf_encoder_decoder.py

```diff
@@ -0,0 +1,671 @@
+        return encoder_model, decoder_model
+
+    def prepare_config_and_inputs(self):
+        model_tester_encoder = TFBertModelTester(self, batch_size=13)
+        model_tester_decoder = TFBertModelTester(self, batch_size=13)
+        encoder_config_and_inputs = model_tester_encoder.prepare_config_and_inputs()
+        decoder_config_and_inputs = model_tester_decoder.prepare_config_and_inputs_for_decoder()
+        (
+            config,
+            input_ids,
+            token_type_ids,
+            attention_mask,
+            sequence_labels,
+            token_labels,
+            choice_labels,
+        ) = encoder_config_and_inputs
+        (
+            decoder_config,
+            decoder_input_ids,
+            decoder_token_type_ids,
+            decoder_attention_mask,
+            decoder_sequence_labels,
+            decoder_token_labels,
+            decoder_choice_labels,
+            encoder_hidden_states,
+            encoder_attention_mask,
+        ) = decoder_config_and_inputs
+
+        # make sure that cross attention layers are added
+        decoder_config.add_cross_attention = True
+        #  disable cache for now
+        decoder_config.use_cache = False
+        return {
+            "config": config,
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+            "decoder_config": decoder_config,
+            "decoder_input_ids": decoder_input_ids,
+            "decoder_token_type_ids": decoder_token_type_ids,
+            "decoder_attention_mask": decoder_attention_mask,
+            "decoder_sequence_labels": decoder_sequence_labels,
+            "decoder_token_labels": decoder_token_labels,
+            "decoder_choice_labels": decoder_choice_labels,
+            "encoder_hidden_states": encoder_hidden_states,
+            "labels": decoder_token_labels,
+        }
+
+    @slow
+    @require_torch
+    def test_bert2bert_summarization(self):
+
+        if is_torch_available():
+            from transformers import EncoderDecoderModel
+
+            tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+
+            """Not working, because pt checkpoint has `encoder.encoder.layer...` while tf model has `encoder.bert.encoder.layer...`
+            (For Bert decoder, there is no issue, because `BertModel` is wrapped into `decoder` as `bert`)
+            model = TFEncoderDecoderModel.from_pretrained("patrickvonplaten/bert2bert-cnn_dailymail-fp16", from_pt=True)
+            """
+
+            # workaround to load from pt
+            _model = EncoderDecoderModel.from_pretrained("patrickvonplaten/bert2bert-cnn_dailymail-fp16")
+            _model.encoder.save_pretrained("./encoder")
+            _model.decoder.save_pretrained("./decoder")
+            model = TFEncoderDecoderModel.from_encoder_decoder_pretrained(
+                "./encoder", "./decoder", encoder_from_pt=True, decoder_from_pt=True
```

Comment:
smart hack! :-) Would be nice though to get it working with `from_pretrained(...)` directly